src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -430,8 +436,20 @@ internal void ProcessDto(ModelBindingContext bindingContext, ComplexModelDto dto
                         out requiredValidator);
 
                     SetProperty(bindingContext, modelExplorer, propertyMetadata, dtoResult, requiredValidator);
+
+                    var dtoValidationNode = dtoResult.ValidationNode;
+                    if (dtoValidationNode == null)
+                    {
+                        // Make sure that irrespective of if the properties of the model were bound with a value,
+                        // create a validation node so that these get validated.
+                        dtoValidationNode = new ModelValidationNode(dtoResult.Key, entry.Key, dtoResult.Model);
+                    }
+
+                    validationNode.ChildNodes.Add(dtoValidationNode);
```

Comment:
Tried doing it, but i would prefer not to do it in ComplexModelDTOModelBinder, since the result is immutable, adding a validation node to it is not possible and we would have to create a new result, I dont like it .. makes it ugly. NOT Fixing.